src/Text.Analyzers/Core/IdentifiersShouldBeSpelledCorrectly.cs

```diff
@@ -359,6 +359,14 @@ void AnalyzeSymbol(SymbolAnalysisContext symbolContext)
                             ReportDiagnosticsForSymbol(typeParameter, RemovePrefixIfPresent('T', typeParameter.Name), symbolContext.ReportDiagnostic);
                         }
 
+                        break;
+                    case IParameterSymbol parameter:
+                        //check if the member this parameter is part of is an override/interface implementation
+                        if (parameter.ContainingSymbol.IsImplementationOfAnyImplicitInterfaceMember() || parameter.ContainingSymbol.IsImplementationOfAnyExplicitInterfaceMember())
+                        {
+                            return;
+                        }
```

Comment:
Thinking about this again, I guess we only need to do that if the parameter of the current method has the same name as the one in the interface. For example:

```csharp
public interface IProgram
{
    void Method(string goodName);
}

public class Program : IProgram
{
    public void Method(string baddName) { } // we should report here, I think.
}
```

Also this code checks interface implementations but not overrides. I guess that is not intentional?

Reply:
it wasn't intentional. I removed the IsOverride at some point without meaning to.
added it back and added unit tests.
I agree the _best _thing to do would be to continue to emit the diagnostic if the parameter names does not match the base/interface.

I originally thought the easiest way to do this was to never emit the diagnostic on overrides if CA2517 was enabled.

alternatively you could run the same analysis that CA2517 does and only emit the diagnostic if the names don't match and it is misspelled. (this would be the most robust way of doing it)